Gradle plugin 6.6.1 introduces conflicting variant "spotless123"

(See also https://github.com/projectnessie/cel-java/issues/185 )

Steps to reproduce:
* Clone https://github.com/projectnessie/cel-java
* Run `./gradlew check` -> failure
* Interestingly: `./gradlew assemble check` works (may want to add `-x jmh` to skip running the microbenchmarks)

Downgrading the spotless plugin (in `settings.gradle.kts`) to 6.6.0 "fixes" the problem.

CEL-Java uses the Gradle `rootProject` for dependency management, as that's the easiest way that also works with dependabot.
So submodules use `platform(rootProject)` for pull in dependency versions.
This started to fail with the Spotless Gradle plugin 6.6.1.

It seems that the temporary configurations ("spotless123...") leak variants that are too similar to "regular" variants and cause the conflict. Probably due to some daemon-leak fix in 6.6.1 that causes (too?) eager configuration/variant creation.

```
$ ./gradlew check -x jmh 
Configuration on demand is an incubating feature.

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':conformance:test'.
> Could not resolve all task dependencies for configuration ':conformance:testRuntimeClasspath'.
   > Could not resolve project :.
     Required by:
         project :conformance
         project :conformance > project :core
         project :conformance > project :generated-pb
      > The consumer was configured to find a runtime of a platform compatible with Java 8, packaged as a jar, preferably optimized for standard JVMs, and its dependencies declared externally. However we cannot choose between the following variants of project ::
          - runtimeElements
          - spotless1316845950
        All of them match the consumer attributes:
          - Variant 'runtimeElements' capability org.projectnessie.cel:cel-java:0.3.4-SNAPSHOT declares a runtime of a platform:
              - Unmatched attributes:
                  - Doesn't say anything about how its dependencies are found (required its dependencies declared externally)
                  - Doesn't say anything about its target Java environment (preferred optimized for standard JVMs)
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about its elements (required them packaged as a jar)
          - Variant 'spotless1316845950' capability org.projectnessie.cel:cel-java:0.3.4-SNAPSHOT declares a component, and its dependencies declared externally:
              - Unmatched attributes:
                  - Doesn't say anything about its component category (required a platform)
                  - Doesn't say anything about its target Java environment (preferred optimized for standard JVMs)
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about its elements (required them packaged as a jar)
                  - Doesn't say anything about its usage (required a runtime)
        The following variants were also considered but didn't match the requested attributes:
          - Variant 'apiElements' capability org.projectnessie.cel:cel-java:0.3.4-SNAPSHOT declares a platform:
              - Incompatible because this component declares an API of a component and the consumer needed a runtime of a component
              - Other compatible attributes:
                  - Doesn't say anything about how its dependencies are found (required its dependencies declared externally)
                  - Doesn't say anything about its target Java environment (preferred optimized for standard JVMs)
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about its elements (required them packaged as a jar)
          - Variant 'enforcedApiElements' capability org.projectnessie.cel:cel-java-derived-enforced-platform:0.3.4-SNAPSHOT:
              - Incompatible because this component declares an API of an enforced platform and the consumer needed a runtime of a platform
              - Other compatible attributes:
                  - Doesn't say anything about how its dependencies are found (required its dependencies declared externally)
                  - Doesn't say anything about its target Java environment (preferred optimized for standard JVMs)
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about its elements (required them packaged as a jar)
          - Variant 'enforcedRuntimeElements' capability org.projectnessie.cel:cel-java-derived-enforced-platform:0.3.4-SNAPSHOT declares a runtime of a component:
              - Incompatible because this component declares an enforced platform and the consumer needed a platform
              - Other compatible attributes:
                  - Doesn't say anything about how its dependencies are found (required its dependencies declared externally)
                  - Doesn't say anything about its target Java environment (preferred optimized for standard JVMs)
                  - Doesn't say anything about its target Java version (required compatibility with Java 8)
                  - Doesn't say anything about its elements (required them packaged as a jar)
   > Could not find com.google.protobuf:protobuf-java:.
     Required by:
         project :conformance
         project :conformance > project :generated-pb
   > Could not find io.grpc:grpc-protobuf:.
     Required by:
         project :conformance
   > Could not find io.grpc:grpc-stub:.
     Required by:
         project :conformance
   > Could not find io.grpc:grpc-netty-shaded:.
     Required by:
         project :conformance
   > Could not find org.assertj:assertj-core:.
     Required by:
         project :conformance
   > Could not find org.junit.jupiter:junit-jupiter-api:.
     Required by:
         project :conformance
   > Could not find org.junit.jupiter:junit-jupiter-params:.
     Required by:
         project :conformance
   > Could not find org.junit.jupiter:junit-jupiter-engine:.
     Required by:
         project :conformance
   > Could not find org.agrona:agrona:.
     Required by:
         project :conformance > project :core
```

--- a/plugin-gradle/CHANGES.md
+++ b/plugin-gradle/CHANGES.md
@@ -3,6 +3,8 @@
 We adhere to the [keepachangelog](https://keepachangelog.com/en/1.0.0/) format (starting after version `3.27.0`).
 
 ## [Unreleased]
+### Fixed
+* Prevent tool configurations from being resolved outside project ([#1447](https://github.com/diffplug/spotless/pull/1447))
 
 ## [6.12.1] - 2023-01-02
 ### Fixed

--- a/plugin-gradle/CHANGES.md
+++ b/plugin-gradle/CHANGES.md
@@ -4,7 +4,7 @@ We adhere to the [keepachangelog](https://keepachangelog.com/en/1.0.0/) format (
 
 ## [Unreleased]
 ### Fixed
-* Prevent tool configurations from being resolved outside project ([#1447](https://github.com/diffplug/spotless/pull/1447))
+* Prevent tool configurations from being resolved outside project ([#1447](https://github.com/diffplug/spotless/pull/1447) fixes [#1215](https://github.com/diffplug/spotless/issues/1215))
 
 ## [6.12.1] - 2023-01-02
 ### Fixed

--- a/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/GradleProvisioner.java
+++ b/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/GradleProvisioner.java
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016-2022 DiffPlug
+ * Copyright 2016-2023 DiffPlug
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/GradleProvisioner.java
+++ b/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/GradleProvisioner.java
@@ -118,6 +118,7 @@ class GradleProvisioner {
 						.forEach(config.getDependencies()::add);
 				config.setDescription(mavenCoords.toString());
 				config.setTransitive(withTransitives);
+				config.setCanBeConsumed(false);
 				config.attributes(attr -> {
 					attr.attribute(Bundling.BUNDLING_ATTRIBUTE, project.getObjects().named(Bundling.class, Bundling.EXTERNAL));
 				});

--- a/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/GradleProvisioner.java
+++ b/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/GradleProvisioner.java
@@ -119,6 +119,7 @@ class GradleProvisioner {
 				config.setDescription(mavenCoords.toString());
 				config.setTransitive(withTransitives);
 				config.setCanBeConsumed(false);
+				config.setVisible(false);
 				config.attributes(attr -> {
 					attr.attribute(Bundling.BUNDLING_ATTRIBUTE, project.getObjects().named(Bundling.class, Bundling.EXTERNAL));
 				});

